src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java

```diff
@@ -23,7 +33,23 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throw new RegistrationException("User with email "
                     + requestDto.getEmail() + " already exists");
         }
-        User savedUser = userRepository.save(userMapper.toUser(requestDto));
+        requestDto.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+        User userWithEncodedPassword = userMapper.toUser(requestDto);
+        User savedUser = userRepository.save(userWithEncodedPassword);
         return userMapper.toUserResponseDto(savedUser);
     }
+
+    @Override
+    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
```

Comment:
Let's create AuthenticationService in security package and replace authentication logic there
```suggestion
    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
        final Authentication authentication = manager
                .authenticate(
                        new UsernamePasswordAuthenticationToken(
                                requestDto.email(), requestDto.password()));
        String token = jwtUtil.generateToken(authentication.getName());
        return new UserLoginResponseDto(token);
    }
```

Also, add bean to SecurityConfig
```
@Bean
    public AuthenticationManager authenticationManager(
            AuthenticationConfiguration authenticationConfiguration) throws Exception {
        return authenticationConfiguration.getAuthenticationManager();
    }
```